sample-schemas/demo_schema.erl

```diff
@@ -0,0 +1,66 @@
+-module(demo_schema).
+
+-include_lib("typerefl/include/types.hrl").
+
+-type ip() :: ipv4() | ipv6().
+-type ipv4() :: {0..255, 0..255, 0..255, 0..255}.
+-type ipv6() :: {0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535}.
+
+-reflect_type([ip/0]).
+
+-export([keys/0]).
+
+keys() ->
+    [ {"foo.setting", fun foo__setting/1}
```

Comment:
we do not need to follow `__` for `.` naming convention here
that was for the old proposal which we try to get field names from exported functions list.